### PR TITLE
Progress bars to bootstrapping and making the bootstrap tables more consistent

### PR DESCRIPTION
**Pull request by Kucharssim**
https://github.com/jasp-stats/jasp-test-release/issues/168

1) Adds progress bars to bootstrapping:
- coefficients of linear regression
- coefficients of logistic regression
- marginal means and post hoc in AN(C)OVA
- marginal means in RM ANOVA

2) Point estimates from bootstrapping in linear and logistic reg. no longer report the original estimate, but the median of the bootstrap

3) Adds some footnotes to bootstrapped tables in linear and logistic reg. to make the bootstrapping style more consistent.

### CODE REVIEW

**Comment by TimKDJ** [comment]
Looks nice so far! Can you fix the linear regression state so it doesn't resample if you tick the box "confidence interval"? Behaviour is then also more inline with the anova's.

Optional: it might be good to add a way to abort bootstrapping; it can take a while and at the moment you have to wait for it to finish. If you're interested in adding this then you can look [here](https://github.com/jasp-stats/jasp-desktop/commit/7cad3f60cd1ecac96f4d1b8efc7f920435d31510#diff-bd2e70b582b29699c2bee513530f9eb4) for some pointers. The progressbar can return changed options so it's a two birds one stone kinda thing.

**Comment by boutinb** [comment]
Hi @Kucharssim Will you implement the suggestions this week? We would like to finish the fixes for the release this week.

**Comment by Kucharssim** [comment]
Hi @boutinb, yes I am planning to get this done this week

File: JASP-Engine/JASP/R/ancova.R [hunk 1]
```diff
@@ -1511,6 +1520,12 @@ Ancova <- function(dataset=NULL, options, perform="run", callback=function(...)
       
       # actual bootstrapping
       .bootstrapPostHoc <- function(data, indices, options, thisVarName, postHocVariablesListV, postHocVarIndex){
+        pr <- progress()
+        response <- .callbackBootstrapAncovaPostHoc(pr, options)
```
[the remark this answers is not in the document]
Comment (Kucharssim) [reply JASP-Engine/JASP/R/ancova.R]: Yeah, I have tried this, but I could not get the options from the environment in for some reason (the method implemented in the commit you referenced to earlier did not work inside the boot function).

**Comment by Kucharssim** [comment]
> It works for the most part, but there are some errors when you abort bootstrapping:
> 
> Remove RM factor after the bootstrapping started:
> Screenshot 2019-06-04 at 12 12 25
> 

Ah yeah, I will fix this tomorrow.

> Remove a model term after the bootstrapping started:
> Screenshot 2019-06-04 at 12 28 02

This is probably caused rather by the small number of resamples (100) than the aborting - I guess it would be good to add some checks if the CIs can be computed.

**Comment by boutinb** [comment]
@Kucharssim Can you make this change as soon as possible? This is the fix we need for the release.

**Comment by TimKDJ** [comment]
@Kucharssim I see you pushed changes -- are these final? Then I'll go test

**Comment by Kucharssim** [comment]
Yes you can test - I fixed the things you found and make the jumping out of bootstrap more specific as you asked - however I only renamed the "callback" functions

**Pull request merged**